thunderfish/chirp.py

```diff
@@ -0,0 +1,196 @@
+"""
+Detection of chirps in weakly electric fish recordings.
+
+chirp_analysis(): calculates spectrogram, detects fishes and extracts chirp times(combined for all fishes).
+                  !!! recommended for short recordings (up to 5 min) where only the chirp times shall be extracted !!!
+chirp_detection(): extracts chirp times with help of given spectrogram and fishlist.
+chirp_spectrogram(): performs spectrogram suitable for chip detection.
+"""
+
+import sys
+import numpy as np
+import matplotlib.mlab as mlab
+import dataloader as dl
+import harmonicgroups as hg
+import config_tools as ct
+import powerspectrum as ps
+import matplotlib.pyplot as plt
+import peakdetection as pkd
+
+def chirp_spectrogram(data, samplerate, fresolution=0.5, detrend=mlab.detrend_none, window=mlab.window_hanning, overlap_frac=0.5,
+                      pad_to=None, sides='default', scale_by_freq=None):
+    """
```

Comment:
This definitely is a function for the powerspectrum module! There are many use cases for a spectrogram, not only chirp detection!